src/components/Modals/PublishWizardCollectionModal/ConfirmCollectionItemsStep/ConfirmCollectionItemsStep.tsx

```diff
@@ -0,0 +1,106 @@
+import { ethers } from 'ethers'
+
+const HUGE_PAGE_SIZE = 20
+
+export const ConfirmCollectionItemsStep: React.FC<Pick<Props, 'collection'> & { onNextStep: () => void }> = props => {
+  const { collection, onNextStep } = props
+
+  const renderPrice = (item: Item) => {
+    return (
+      <div>
+        {isFree(item) ? (
+          t('global.free')
+        ) : (
+          <Mana className="mana" network={Network.MATIC}>
+            {ethers.utils.formatEther(item.price!)}
+          </Mana>
+        )}
+      </div>
+    )
+  }
+
+  const renderItemsTable = (items: Item[]) => {
+    return (
+      <Table basic="very" columns={4}>
+        <Table.Header>
+          <Table.HeaderCell width={5}>{t('collection_detail_page.table.item')}</Table.HeaderCell>
+          <Table.HeaderCell>{t('collection_detail_page.table.rarity')}</Table.HeaderCell>
+          <Table.HeaderCell>{t('collection_detail_page.table.category')}</Table.HeaderCell>
+          <Table.HeaderCell>{t('collection_detail_page.table.price')}</Table.HeaderCell>
+        </Table.Header>
+        <Table.Body>
+          {items.map(item => (
+            <Table.Row key={item.id} className="CollectionItem row">
+              <Table.Cell className="avatarColumn">
+                <div className="avatarContainer">
+                  <ItemImage className="itemImage" item={item} hasRarityBackground={false} />
+                  <div className="info">
+                    <div className="name" title={item.name}>
+                      {item.name}
+                    </div>
+                    <ItemBadge className="badge" item={item} size="small"></ItemBadge>
+                  </div>
+                </div>
+              </Table.Cell>
+              <Table.Cell className="column">
+                <RarityBadge category={item.data.category!} rarity={item.rarity!} size="small" />
+              </Table.Cell>
+              <Table.Cell className="column">
+                <div>{t(`${item.type}.category.${item.data.category}`)}</div>
+              </Table.Cell>
+              <Table.Cell className="column priceColumn">{renderPrice(item)}</Table.Cell>
+            </Table.Row>
+          ))}
+        </Table.Body>
+      </Table>
+    )
+  }
+
+  return (
+    <>
+      <Modal.Content className="ConfirmCollectionItemsStep">
+        <Column>
+          <Row className="details">
+            <Column grow={true}>
+              <p className="title">{t('publish_collection_modal_with_oracle.confirm_collection_items_step.title')}</p>
+              <p className="subtitle">
+                {t('publish_collection_modal_with_oracle.confirm_collection_items_step.subtitle', { enter: <br /> })}
+              </p>
+              <p className="description">{t('publish_collection_modal_with_oracle.confirm_collection_items_step.description')}</p>
+              <div className="items">
+                <CollectionProvider id={collection?.id} itemsPage={1} itemsPageSize={HUGE_PAGE_SIZE}>
```

Comment:
Since this is opened from the collection detail page, I wonder if we can avoid using this provider here and get the collection and their items directly from the state. What do you think?

Reply:
thanks for the feedback @juanmahidalgo! makes sense to get the data directly from the state. I'll be updating the PR soon